### ISSUE TITLE
Consider renaming DataTree.ds and/or the `data` argument in DataTree.__init__

### ISSUE THREAD

**Issue by shoyer** [pydata/xarray#9458]
### What is your issue?

It is not inherently clear to that `DataTree(data=...)` and `DataTree.ds` refer to the same thing, or that it refers specifically to the dataset associated with the node. We also do not use the abbreviation `.ds` anywhere else in Xarray.

Instead, what about calling this argument `node`? This makes the data model of DataTree much more transparent, it's just a `node` and `children`, with optional `name` and `parent`.

Example usage:
```python
tree = DataTree(node=Dataset(...))
tree.node = Dataset(...)
```

CC @flamingbear @TomNicholas @eni-awowale @owenlittlejohns

**Comment by TomNicholas** [comment]
> It is not inherently clear to that DataTree(data=...) and DataTree.ds refer to the same thing, or that it refers specifically to the dataset associated with the node.

That's a fair criticism.

> We also do not use the abbreviation `.ds` anywhere else in Xarray.

We don't in the API, but we do everywhere in the examples, e.g. `ds = open_dataset(...)`. I agree that the current situation of `.ds` and `.to_dataset()` is not ideal though.

> Instead, what about calling this argument node? This makes the data model of DataTree much more transparent, it's just a node and children, with optional name and parent.

**Comment by TomNicholas** [comment]
I think I'm way too deep in the weeds here, and would appreciate perspective from @pydata/xarray on what they think the most intuitive names for these parts of the tree would be. Now is the only time we can ever change this really.

**Comment by shoyer** [comment]
Did we ever write the doc on DataTree terminology? That could be a good place to sort this all out.

**Comment by dcherian** [comment]
How about simply `.dataset`? `DataTree(dataset=...)` seems pretty readable to me.

PS: `tree.this` struck me as nice but `DataTree(this=...)` is nonsensical haha.

**Comment by TomNicholas** [comment]
> How about simply `.dataset`? 

That makes the intent quite clear, but 

1. It's a lot longer than `.ds` (or `.node`), and we expect this method to be called a lot,

2. We also have `.to_dataset()`, which is different from `.ds` in that the former returns a mutable copy of the data and the latter returns an [immutable view](https://github.com/pydata/xarray/blob/cc74d3ae226594db23730279711d4379debcf9a8/xarray/core/datatree.py#L157).

> `tree.this`

:-1: to this lol

**Comment by shoyer** [comment]
Thanks @dcherian for the suggestions.

In the DataTree meeting today, we agreed to switch both names to `.dataset`. We'll leave the `.ds` property around for now as a (soft deprecated) alias.

**Comment by TomNicholas** [comment]
Closed by #9476
